### DISCUSSION TITLE
Automatic search for similar links

### DISCUSSION THREAD

**Discussion by erlend-sh** [raffomania/linkblocks#45] [Ideas]
So in talking about single-user AP instances, I just linked someone to this:

https://jvns.ca/blog/2023/08/11/some-notes-on-mastodon/

I know there's a handful other articles just like that which I've read, detailing the 'single user' experience. I just don't have those links handy.

So not only would I like to put this on a "single-person activitypub" [list](https://github.com/raffomania/linkblocks/discussions/37) that others can add similar articles to, but that list itself would ideally provide a 'similar articles' search for me based on that article alone, eventually based on the other articles added to it as well.

This might necessarily be a premium feature of some kind though, because it would need to tap into a paid service like [Spiderweb](https://spiderwebai.xyz/) or [Kagi](https://help.kagi.com/kagi/api/overview.html).

Related:
- https://addons.mozilla.org/en-US/firefox/addon/similarsites/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search
- https://similarsites.com/

**Comment by raffomania** [comment]
Yeah, there's lots of interesting ways to approach this. At the moment, the only thing I'm sure about is that I won't start by trying to solve it with AI :grimacing:
